lib/diaspora_federation/entities/account_migration.rb

```diff
@@ -44,30 +65,44 @@ def self.from_hash(*args)
 
       private
 
-      # @see DiasporaFederation::Entities::Signable#signature_data
-      def signature_data
-        to_s
+      def old_identity_prop
+        @old_identity
+      end
+
+      def validate
```

Comment:
I would just use `rule :old_identity, :not_empty` in the normal validator, which would work, since we always return something, even when there wasn't anything in the hash or received xml. We only return nothing if the author is the new identity and the old should be set here, so it works as we need it. We would need to replace `optional: true` with `default: nil` (which I think is OK, since it's only half-optional anyway).

Reply:
Maybe we can validate it with `diaspora_id` rule then?
```ruby
rule :old_identity, :diaspora_id
```

Reply:
Oh yeah, of course, that makes sense. :+1: